### PR TITLE
HBASE-25627: [Backport]HBase replication should have a metric to represent if the source is stuck getting initialized

### PR DESCRIPTION
**Pull request by sandeepvinayak**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 11s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2.3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 33s |  branch-2.3 passed  |
| +1 :green_heart: |  compile  |   3m 57s |  branch-2.3 passed  |
| +1 :green_heart: |  checkstyle  |   1m 33s |  branch-2.3 passed  |
| +1 :green_heart: |  spotbugs  |   2m 55s |  branch-2.3 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 11s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 56s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 56s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  The patch passed checkstyle in hbase-hadoop-compat  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  The patch passed checkstyle in hbase-hadoop2-compat  |
| +1 :green_heart: |  checkstyle  |   1m  5s |  hbase-server: The patch generated 0 new + 0 unchanged - 11 fixed = 0 total (was 11)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  16m 52s |  Patch does not cause any errors with Hadoop 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  spotbugs  |   3m 25s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 38s |  The patch does not generate ASF License warnings.  |
|  |   |  51m 24s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3076/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3076 |
| JIRA Issue | HBASE-25627 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 7da4b6845bc7 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2.3 / b6d4c9e114 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 96 (vs. ulimit of 12500) |
| modules | C: hbase-hadoop-compat hbase-hadoop2-compat hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3076/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2.3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 25s |  branch-2.3 passed  |
| +1 :green_heart: |  compile  |   1m 49s |  branch-2.3 passed  |
| +1 :green_heart: |  shadedjars  |   5m 56s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 20s |  branch-2.3 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 42s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 42s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 45s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 18s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 28s |  hbase-hadoop-compat in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 30s |  hbase-hadoop2-compat in the patch passed.  |
| +1 :green_heart: |  unit  | 128m 31s |  hbase-server in the patch passed.  |
|  |   | 159m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3076/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3076 |
| JIRA Issue | HBASE-25627 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 09b777af3999 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2.3 / b6d4c9e114 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3076/1/testReport/ |
| Max. process+thread count | 4021 (vs. ulimit of 12500) |
| modules | C: hbase-hadoop-compat hbase-hadoop2-compat hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3076/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 16s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2.3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 58s |  branch-2.3 passed  |
| +1 :green_heart: |  compile  |   1m 29s |  branch-2.3 passed  |
| +1 :green_heart: |  shadedjars  |   5m 29s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  7s |  branch-2.3 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 36s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 28s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 28s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 28s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  2s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 25s |  hbase-hadoop-compat in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 27s |  hbase-hadoop2-compat in the patch passed.  |
| -1 :x: |  unit  | 204m 12s |  hbase-server in the patch failed.  |
|  |   | 232m 46s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3076/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3076 |
| JIRA Issue | HBASE-25627 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 4cd37a33b0cd 4.15.0-126-generic #129-Ubuntu SMP Mon Nov 23 18:53:38 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2.3 / b6d4c9e114 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3076/1/artifact/yetus-jdk8-hadoop2-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3076/1/testReport/ |
| Max. process+thread count | 2899 (vs. ulimit of 12500) |
| modules | C: hbase-hadoop-compat hbase-hadoop2-compat hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3076/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by sandeepvinayak** [comment]
FYI @bharathv 
failing test is unrelated

File: hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java [hunk 1]
```diff
@@ -113,6 +111,8 @@
   private int maxRetriesMultiplier;
   // Indicates if this particular source is running
   volatile boolean sourceRunning = false;
+  // Indicates if the source initialization is in progress
+  private volatile boolean startupOngoing = false;
```
Comment (bharathv) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java:115]: This is not accessed anywhere?

Should be backport HBASE-24877 and then apply this change?

**Review by bharathv — CHANGES_REQUESTED** [review]

File: hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java (hunk 1 above)
Comment (sandeepvinayak) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java:115]: @bharathv yes I see that this was not backported to branch2.3, do you mean creating a new Jira for backporting or can you commit it without jira? I can then merge the latest changes.

Comment (bharathv) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java:115]: Ya, let me try backporting it and then you can rebase. will update you once it is committed.

**Comment by bharathv** [comment]
Closing this PR, it looks like the branches diverged quite a bit and this is not critical fix, so perhaps not worth the effort.

**Pull request closed without merging**